src/js/nettest.js

```diff
@@ -91,7 +96,13 @@ function gatherCandidates(config, params, isGood) {
       }
     } else {
       pc.close();
-      reportFatal('Failed to gather specified candidates');
+      if (!params.optional[0].googIPv6) {
+        reportWarning('Failed to gather IPv6 candidates, it ' +
+          'might not be setup/supported on the network.');
```

Comment:
you are missing many setTestFinished() in the places where you replace reportFatal with reportError/Warning.